backend/api/activity/index.js

```diff
@@ -6,68 +6,224 @@ const activity = express.Router();
     try {
+	  //Save it to the database
       await activitydata.save();
+	  
+	  //Response
       return res.status(statusCode).json({ activity: activitydata });
     } catch (error) {
+	  //Error Response
       return res.status(500).send(error);
     }
   });
 
 /**
  * Edits activity by id
  * TODO: Implement this
- * @param {Number} id - The id of the activity, must be only digits
+ * @param {Hex} id - The id of the activity, must be only hex digits
  */
-activity.route('/edit/:id([0-9]+)')
-  .post(passport.authenticate('local'), async (req, res) => {
-    if (!req.user) {
-      return res.status(403).json({
+activity.route('/edit/:id([a-f0-9]+)')
+  .post(async (req, res) => {
+	//Ensure user is authenticated
+	if (!req.user) {
+      return res.status(401).json({
         error: 'Not authenticated',
       });
     }
-    const statusCode = 200;
-    const shortened = Boolean(req.query.shortened);
-    return res.status(statusCode).json({
-      id: req.params.id,
-      status: statusCode,
-      shortened,
-    });
+	
+	try {
+	  //Get the activity to edit
+      const getobj = await Activity.findById(req.params.id).exec();
+	  
+	  //Ensure the authenticated user is author of this activity
+	  if (getobj.postedBy != req.user.id) {
+		return res.status(403).json({
+          error: 'Wrong account',
+        });
+	  }
+	  
+	  //Change the activity data
+      const statusCode = 200;
+      const shortened = Boolean(req.query.shortened);
+	  const currentDateTime = Date.now();
+	  getobj.title = req.body.title || getobj.title;
+	  getobj.description = req.body.description || getobj.description;
+	  getobj.startDateTime = req.body.startDateTime || getobj.startDateTime;
+	  getobj.endDateTime = req.body.endDateTime || getobj.endDateTime;
+	  getobj.lastEditDateTime = currentDateTime;
+	  
+	  //Ensure end time comes after start time
+	  if (getobj.endDateTime < getobj.startDateTime) {
+		return res.status(400).json({
+          error: 'endDateTime is less than startDateTime',
+        });
+	  }
+	  
+	  //Save it to the database
+	  await getobj.save();
+	  
+	  //Response
+	  res.status(statusCode).json({
+        activity: getobj,
+		shortened,
+      });
+    } catch (error) {
+	  //Error Response
+      res.status(500).send(error);
+    }
+  });
+  
+/**
+ * Deletes an activity by id
+ * @param {Hex} id - The id of the activity, must be only hexadecimal
+ */
+activity.route('/delete/:id([a-f0-9]+)')
+  .post(async (req, res) => {
+	//Ensure user is authenticated
+	if (!req.user) {
+      return res.status(401).json({
+        error: 'Not authenticated',
+      });
+    }
+	
+	try {
+	  //Get the activity to edit
+      const getobj = await Activity.findById(req.params.id).exec();
+	  
+	  //Ensure the authenticated user is author of this activity
+	  if (getobj.postedBy != req.user.id) {
+		return res.status(403).json({
+          error: 'Wrong account',
+        });
+	  }
+	  
+	  //Delete it
+	  await Activity.deleteOne(getobj);
```

Comment:
Just a thought, maybe we can have deleted activities be marked as deleted in a flag instead of actually deleted. This could be handy later on for server maintenance or something.